resources/js/components/entries/CollectionCalendar.vue

```diff
@@ -0,0 +1,509 @@
+<script setup>
+import { ref, watch, computed, nextTick } from 'vue';
+import axios from 'axios';
+import { CalendarHeader, CalendarHeading, CalendarRoot } from 'reka-ui';
+import { CalendarDate } from '@internationalized/date';
+import CalendarMonthView from './CalendarMonthView.vue';
+import CalendarWeekView from './CalendarWeekView.vue';
+import { Listing, StatusIndicator } from '@/components/ui';
+import { useCalendarDates } from '@/composables/useCalendarDates';
+
+const props = defineProps({
+    collection: { type: String, required: true },
+    blueprints: { type: Array, default: () => [] },
+    createUrl: { type: String, required: true },
+});
+
+const emit = defineEmits(['changed', 'saved', 'canceled']);
+
+const currentDate = ref(new CalendarDate(new Date().getFullYear(), new Date().getMonth() + 1, new Date().getDate()));
+const selectedDate = ref(null);
+const entries = ref([]);
+const loading = ref(false);
+const error = ref(null);
+const saving = ref(false);
+const viewMode = ref('month'); // 'month' or 'week'
+const weekViewRef = ref(null);
+const pendingDateChanges = ref(new Map()); // Track entry ID -> new date
+const isDirty = ref(false);
+// Reactive drag state for Vue class bindings
+const dragOverTarget = ref(null);
+
+// Use composables
+const {
+    formatDateString,
+    createDateFromCalendarDate,
+    getWeekDates,
+    getVisibleHours,
+    getHourLabel,
+    formatTime,
+    isToday,
+    getCurrentDateRange
+} = useCalendarDates();
+
+// ============================================================================
+// API & Data Management
+// ============================================================================
+
+async function fetchEntries() {
+    loading.value = true;
+    error.value = null;
+
+    try {
+        // Guard against undefined values
+        if (!currentDate.value || !viewMode.value) {
+            console.warn('fetchEntries called with undefined values:', { currentDate: currentDate.value, viewMode: viewMode.value });
+            return;
+        }
+
+        const { startDate, endDate } = getCurrentDateRange(currentDate.value, viewMode.value);
+
+        if (!startDate || !endDate) {
+            console.error('getCurrentDateRange returned undefined values:', { startDate, endDate });
+            return;
+        }
+
+        const response = await axios.get(cp_url(`collections/${props.collection}/entries`), {
+            params: {
```

Comment:
I wonder if we might need to pass down the `filters` object from the controller so the calendar only returns entries from the current site.

We had to do it for the grid view: #12324